### ISSUE TITLE
Possible inconsistencies with LeJEPA paper (loss weighting + SIGReg views) and AI-Frontiers discord application

### ISSUE THREAD

**Issue by kc-111** [rbalestr-lab/stable-pretraining#413]
Hi contributors,

I really like the current repository and am using it over my own code for consistency of results. I am currently trying to reproduce the results in the LeJEPA paper and noticed a couple of things I wanted to flag. 

First, the current implementation uses 
```
loss = inv_loss + lamb * sigreg_loss
``` 
whereas the paper has 
```
loss = (1 - lamb) * inv_loss + lamb * sigreg_loss.
```
This affects hyperparameter transfer and optimizer behavior since the gradient scales differ between the two forms, so published λ values won't carry over directly. 

Second, SIGReg is applied to all views and embeddings flattened together, whereas the paper applies it per view and averages. I wonder which is the preferred approach here? As a thought experiment, the flattened version can be seen as projecting a mixture of Gaussians (one per image, spread by its views) whereas the per-view version is closer to projecting point masses where each sample contributes a single point per test. This may create interesting but potentially very different representations. Neither is obviously wrong, but they do diverge from the paper's setup. As food for thought, I wonder if the view spread (which kind of smooths the ECF in Epps-Pulley) in the flattened version partially confounds the Gaussianity signal early in training. 

I would love to discuss further and about other ideas I have. My Discord application is still pending!

Thanks!
KC

**Comment by alessioarcara** [comment]
interested as well
